### PR TITLE
feat(application-system-api): mounting secrets for payments api

### PR DESCRIPTION
**Pull request by jGeir**
# ...

Attach a link to issue if relevant

## What

We added a new feature to test out payment process for external partners - Little did we know that we needed to mount the secret variables.

## Why

In order for https://beta.dev01.devland.is/umsoknir/borga to work

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by pshomov — APPROVED** [review]
LGTM

File: apps/api/infra/api.ts [hunk 1]
```diff
@@ -174,6 +174,15 @@ export const serviceSetup = (services: {
...
         '/k8s/documentprovider/DOCUMENT_PROVIDER_ADMINS',
+      xRoadBaseUrl: '/k8s/application-system-api/XROAD_BASE_PATH',
+      xRoadProviderId: '/k8s/application-system-api/PAYMENT_XROAD_PROVIDER_ID',
+      xRoadClientId: '/k8s/application-system-api/XROAD_CLIENT_ID',
+      username: '/k8s/application-system-api/PAYMENT_USER',
+      password: '/k8s/application-system-api/PAYMENT_PASSWORD',
+      callbackBaseUrl: '/k8s/application-system-api/PAYMENT_BASE_CALLBACK_URL',
+      callbackAdditionUrl:
+        '/k8s/application-system-api/PAYMENT_ADDITION_CALLBACK_URL',
+      arkBaseUrl: '/k8s/application-system-api/ARK_ENDPOINT',
```
[the remark this answers is not in the document]
Comment (jeremybarbet) [reply apps/api/infra/api.ts:185]: Shouldn't they be Snake Case All Caps as well?

[the remark this answers is not in the document]
Comment (jGeir) [reply apps/api/infra/api.ts]: I've removed these three secrets from API - Cannot remove more. Good catch!

@jeremybarbet Everything else is - is that the best practice within charts?

[the remark this answers is not in the document]
Comment (jeremybarbet) [reply apps/api/infra/api.ts]: I would assume it's the convention to keep consistency across the carts

[the remark this answers is not in the document]
Comment (sindrig) [reply apps/api/infra/api.ts]: definitely - we want all environment variables in snake case caps

[the remark this answers is not in the document]
Comment (jGeir) [reply apps/api/infra/api.ts]: I'll get to it!

File: apps/api/src/app/environments/environment.ts [hunk 2]
```diff
@@ -104,16 +104,16 @@ const devConfig = {
     clientId: process.env.XROAD_CLIENT_ID ?? 'DEV/GOV/10000/island-is-client',
   },
   paymentDomain: {
-    xRoadBaseUrl: process.env.XROAD_BASE_PATH,
-    xRoadProviderId:
+    XROAD_BASE_URL: process.env.XROAD_BASE_PATH,
```
Comment (jeremybarbet) [inline apps/api/src/app/environments/environment.ts:107]: You can keep it in camelCase here as well for the variables

File: apps/application-system/api/src/environments/environment.ts [hunk 3]
```diff
@@ -52,21 +52,21 @@ const devConfig = {
     presignBucket: process.env.FILE_SERVICE_PRESIGN_BUCKET,
     attachmentBucket: process.env.APPLICATION_ATTACHMENT_BUCKET,
     paymentOptions: {
-      arkBaseUrl: process.env.ARK_ENDPOINT,
-      xRoadBaseUrl: process.env.XROAD_BASE_PATH ?? 'http://localhost:8081',
-      xRoadClientId:
+      ARK_BASE_URL: process.env.ARK_ENDPOINT,
```
Comment (jeremybarbet) [inline apps/application-system/api/src/environments/environment.ts:55]: It will stays camelCase as well here

**Review by jeremybarbet — CHANGES_REQUESTED** [review]
You converted a bit too many variables into `snake case`. In short, infra required snake case and apps/libs on the development side is requiring `camel case`

The only changes needed to `snake case` is within the `apps/api/infra/api.ts` file

**Review by jeremybarbet — APPROVED** [review]

**Pull request merged**